### PR TITLE
✨ added check for duplicated keys -#78843

### PR DESCRIPTION
**Pull request by AGr-AlexandrGrigoryev**
Added check of duplicated keys in the Spreadsheet before writing to the file. If duplicate keys exist, show the error with duplicated keys.
#### Checklist
- [ ] Added tests (if applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 ## master
 
 ### Added
+- Add check for duplicated keys in Spreadsheet ([#38](https://github.com/AckeeCZ/ACKLocalization/pull/38), kudos to @AGr-AlexandrGrigoryev)
```
Comment (LukasHromadnik) [inline CHANGELOG.md]: please rename it to "duplicate keys" 🤓

File: Sources/ACKLocalizationCore/ACKLocalization.swift [hunk 2]
```diff
@@ -413,22 +413,22 @@ public final class ACKLocalization {
     private func writeRows(_ rows: [LocRow], to file: String) throws {
         guard rows.count > 0 else { return }
 
-        try checkDuplicateKeys(form: rows)
+        try checkDuplicatedKeys(form: rows)
```
Comment (LukasHromadnik) [inline Sources/ACKLocalizationCore/ACKLocalization.swift]: "checkDuplicateKeys" was correct

File: Sources/ACKLocalizationCore/ACKLocalization.swift [hunk 3]
```diff
@@ -412,12 +412,26 @@ public final class ACKLocalization {
...
         try rows.map { $0.localizableRow }
             .joined(separator: "\n")
             .write(toFile: file, atomically: true, encoding: .utf8)
     }
-    
+
+    /// Check if given `rows` have a duplicated keys
+    public func checkDuplicatedKeys(form rows: [LocRow]) throws {
+        let keys = rows.map { $0.key }
+        let uniqueKeys = Set(rows.map { $0.key })
```
Comment (LukasHromadnik) [inline Sources/ACKLocalizationCore/ACKLocalization.swift]: 💡 `let uniqueKeys = Set(keys)`

File: Tests/ACKLocalizationCoreTests/LocRowTests.swift [hunk 4]
```diff
@@ -81,4 +81,14 @@ final class LocRowTests: XCTestCase {
         let locRow = LocRow(key: "pos_arg_key", value: "%1$@ people will arrive in %2$@ minutes")
         XCTAssertEqual(#""pos_arg_key" = "%1$@ people will arrive in %2$@ minutes";"#, locRow.localizableRow)
     }
+
+    func testForDuplicatedKeys() throws {
```
Comment (LukasHromadnik) [inline Tests/ACKLocalizationCoreTests/LocRowTests.swift]: please move it to the `ACKLocalizationTests`, this has nothing to do with `LocRow`. Also add test for success path (one duplicate keys)

**Review by LukasHromadnik — CHANGES_REQUESTED** [review]

**Review by LukasHromadnik — APPROVED** [review]
